src/nvim/shada.c

```diff
@@ -2216,8 +2222,22 @@ static inline ShaDaWriteResult shada_read_when_writing(ShaDaReadDef *const sd_re
               }
               shada_free_shada_entry(&wms_entry->data);
             }
+          } else {
+            FOR_ALL_BUFFERS(buf) {
+              if (strequal((char *)buf->b_ffname, entry.data.filemark.fname)) {
+                fmark_T *fmp = xmalloc(sizeof(fmark_T));
+                mark_get(buf, curwin, fmp, kMarkBufLocal, (int)entry.data.filemark.name);
+                if (fmp->timestamp > entry.timestamp) {
+                  set_wms = false;
+                  break;
+                }
+                XFREE_CLEAR(fmp);
```

Comment:
`fmark_T` can be allocated on the stack here:
```suggestion
                fmark_T fmp;
                mark_get(buf, curwin, &fmp, kMarkBufLocal, (int)entry.data.filemark.name);
                if (fmp.timestamp > entry.timestamp) {
                  set_wms = false;
                  break;
                }
```